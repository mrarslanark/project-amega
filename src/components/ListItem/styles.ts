import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    height: 24,
  },
  textWrapper: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
});
