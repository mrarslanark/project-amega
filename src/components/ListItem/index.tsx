import React from 'react';
import {View} from 'react-native';
import {TradeDetail} from '../../screens/MarketData';
import Text from '../Text';
import {styles} from './styles';
import dayjs from 'dayjs';

interface ListItemProps extends TradeDetail {
  useAsHeader?: boolean;
}

const ListItem: React.FC<ListItemProps> = ({
  useAsHeader = false,
  eventTime,
  is_market_maker,
  price,
  quantity,
}): React.JSX.Element | null => {
  if ((!eventTime || !price || !quantity) && !useAsHeader) {
    return null;
  }

  const dollars = new Intl.NumberFormat('en-US', {
    currency: 'USD',
    style: 'currency',
  });

  return (
    <View style={styles.wrapper}>
      <View style={styles.textWrapper}>
        <Text variant="medium">
          {useAsHeader ? 'Event Time' : dayjs(eventTime).format('hh:mm:ss')}
        </Text>
      </View>
      <View style={styles.textWrapper}>
        <Text variant="medium">
          {useAsHeader ? 'Market Maker' : is_market_maker ? 'Yes' : 'No'}
        </Text>
      </View>
      <View style={styles.textWrapper}>
        <Text variant="medium">
          {useAsHeader ? 'Price Change' : dollars.format(Number(price))}
        </Text>
      </View>
      <View style={styles.textWrapper}>
        <Text variant="medium">{useAsHeader ? 'Quantity' : quantity}</Text>
      </View>
    </View>
  );
};

export default ListItem;
