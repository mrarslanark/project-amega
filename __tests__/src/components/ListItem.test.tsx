import {render, screen} from '@testing-library/react-native';
import React from 'react';
import ListItem from '../../../src/components/ListItem';

describe('ListItem', () => {
  it('should render successfully', () => {
    render(
      <ListItem
        eventType="Event Type"
        price="100"
        quantity="Quantity"
        symbol="BTCUSDT"
      />,
    );
    expect(screen).toMatchSnapshot();
  });

  it('should render null if any of the props is undefined', () => {
    render(<ListItem eventType="Event Type" price="100" quantity="Quantity" />);
    expect(screen).toMatchSnapshot();
  });
});
