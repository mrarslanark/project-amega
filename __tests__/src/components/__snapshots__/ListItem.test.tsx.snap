// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render null if any of the props is undefined 1`] = `null`;

exports[`ListItem should render successfully 1`] = `null`;
